client/components/pie-chart/legend.js

```diff
@@ -0,0 +1,63 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import DataType from './data/type';
+import LegendItem from './legend-item';
+import { sortDataAndAssignSections, isDataEqual } from './data';
+
+class PieChartLegend extends Component {
+	static propTypes = {
+		data: PropTypes.arrayOf( DataType ).isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = this.processData( props.data );
+	}
```

Comment:
There's a nice shorthand using property initialisers that should allow us to drop the explicit `constructor` definition:

```jsx
class PieChartLegend extends Component {
	static propTypes = {
		data: PropTypes.arrayOf( DataType ).isRequired,
	};

	state = this.processData( this.props.data );

	/* ... */
}
```

See https://babeljs.io/blog/2015/06/07/react-on-es6-plus, which is sort of our de-facto ES6+ style guide too.